README.md
# export-to-excel

导出数据到excel中

## Super simple to use test

```
var exportToExcel = require('export-to-excel');

var sampleData = [
  {
    "subscribe": 1,
    "openid": "o7Zv3s_F0N3-h09fLN3jOIM8WNwzsdsdsdo",
    "nickname": "name1",
    "sex": 1,
    "language": "zh_CN",
    "city": "杭州",
    "province": "浙江",
    "country": "中国",
    "headimgurl": "http://wx.qlogo.cn/mmopen/s3NiblUuUDR7y3s1DsZibAja25icOumEM4KM79w8pKB5g0o2KKvVDWAqtVuCNVicZIzcqWzOS32ueOvD7tjmRVj2zQ/0",
    "subscribe_time": 1439719607
  },
  {
    "subscribe": 1,
    "openid": "o7Zv3sz92svh2lv_mMg1wewejY0OpU3Q8",
    "nickname": "name2",
    "sex": 1,
    "language": "zh_CN",
    "city": "南京",
    "province": "江苏",
    "country": "中国",
    "headimgurl": "http://wx.qlogo.cn/mmopen/PiajxSqBRaELAI1aEUyI3lwJdMwibicvlkF8ASmIhicSYg3n29v2yHibmum2ibmvedvuXnrziaBl46mnrZe6Cb4pSMaXw/0",
    "subscribe_time": 1431691451
  },
  {
    "subscribe": 0,
    "openid": "o7Zv3s5yjT2MDIICMZkcvcvcvLG71dyBDlg",
    "nickname": "name3",
    "sex": 1,
    "language": "zh_CN",
    "city": "浦东新区",
    "province": "上海",
    "country": "中国",
    "headimgurl": "http://wx.qlogo.cn/mmopen/PiajxSqBRaELt5V5lD4ficPFvT2Z0ZDOHKc26BHh43NXT41WKFQUzLcdtgvBWn1jcqDSac1ib8PpsezuicNVVcbcicA/0",
    "subscribe_time": 1442406029
  }
];


exportToExcel.exportXLSX({
    filename: '微信粉丝列表(2017-01-01)',
    sheetname: '微信粉丝列表',
    title: [
        {
            "fieldName": "subscribe",
            "displayName": "是否关注",
            "cellWidth": 8,
            "type": "bool"  // 1:是  0:否
        },
        {
            "fieldName": "openid",
            "displayName": "OpenID",
            "cellWidth": 30
        },
        {
            "fieldName": "nickname",
            "displayName": "昵称",
            "cellWidth": 15
        },
        {
            "fieldName": "sex",
            "displayName": "性别",
            "cellWidth": 6,
            "type": "sex"  // 1:男  0:女
        },
        {
            "fieldName": "language",
            "displayName": "语言",
            "cellWidth": 8
        },
        {
            "fieldName": "city",
            "displayName": "城市",
            "cellWidth": 12
        },
        {
            "fieldName": "province",
            "displayName": "省",
            "cellWidth": 10
        },
        {
            "fieldName": "country",
            "displayName": "国家",
            "cellWidth": 10
        },
        {
            "fieldName": "headimgurl",
            "displayName": "头像",
            "cellWidth": 20
        },
        {
            "fieldName": "subscribe_time",
            "displayName": "关注时间",
            "cellWidth": 20,
            "type": "datetime"   // 2015-12-12 10:00:00
        }
    ],
    data: sampleData
})
```


## Conversion src to lib

```bash
gulp
```

## Bump up version

```
npm version <version>
```


## Resources

- [export-excel-test](https://github.com/tlqylzn/export-excel-test)
- [SheetJS](https://github.com/SheetJS/js-xlsx)
- [Examples](https://github.com/tlqylzn/export-excel-test)


CHANGELOG.md
<a name="0.2.1"></a>
## 0.2.1 (2017-03-20)




<a name="0.1.11"></a>
## 0.1.11 (2016-01-08)




<a name="0.1.10"></a>
## 0.1.10 (2016-01-07)




<a name="0.1.9"></a>
## 0.1.9 (2016-01-07)




<a name="0.1.9"></a>
## 0.1.9 (2016-01-07)




<a name="0.1.8"></a>
## 0.1.8 (2016-01-05)




<a name="0.1.7"></a>
## 0.1.7 (2016-01-05)




<a name="0.1.6"></a>
## 0.1.6 (2016-01-05)




<a name="0.1.5"></a>
## 0.1.5 (2015-12-31)




<a name="0.1.4"></a>
## 0.1.4 (2015-12-31)




<a name="0.1.3"></a>
## 0.1.3 (2015-12-31)




<a name="0.1.2"></a>
## 0.1.2 (2015-12-30)




<a name="0.1.1"></a>
## 0.1.1 (2015-12-30)






sample.js
module.exports = [
  {
    "subscribe": 1,
    "openid": "o7Zv3s_F0N-h09fLN3jOIM8WNwzo",
    "nickname": "黎叔",
    "sex": 1,
    "language": "zh_CN",
    "city": "杭州",
    "province": "浙江",
    "country": "中国",
    "headimgurl": "http://wx.qlogo.cn/mmopen/s3NiblUuUDR7y3s1DsZibAja25icOumEM4KM79w8pKB5g0o2KKvVDWAqtVuCNVicZIzcqWzOS32ueOvD7tjmRVj2zQ/0",
    "subscribe_time": 1439719607,
    "remark": "",
    "groupid": 0
  },
  {
    "subscribe": 1,
    "openid": "o7Zv3sz9svh2lv_mMg1jY0OpU3Q8",
    "nickname": "石庆年",
    "sex": 1,
    "language": "zh_CN",
    "city": "",
    "province": "",
    "country": "中国",
    "headimgurl": "http://wx.qlogo.cn/mmopen/PiajxSqBRaELAI1aEUyI3lwJdMwibicvlkF8ASmIhicSYg3n29v2yHibmum2ibmvedvuXnrziaBl46mnrZe6Cb4pSMaXw/0",
    "subscribe_time": 1431691451,
    "remark": "",
    "groupid": 101
  },
  {
    "subscribe": 0,
    "openid": "o7Zv3s5yjTMDIICMZkLG71dyBDlg",
    "nickname": "王魁",
    "sex": 1,
    "language": "zh_CN",
    "city": "浦东新区",
    "province": "上海",
    "country": "中国",
    "headimgurl": "http://wx.qlogo.cn/mmopen/PiajxSqBRaELt5V5lD4ficPFvT2Z0ZDOHKc26BHh43NXT41WKFQUzLcdtgvBWn1jcqDSac1ib8PpsezuicNVVcbcicA/0",
    "subscribe_time": 1442406029,
    "remark": "",
    "groupid": 0
  },
  {
    "subscribe": 1,
    "openid": "o7Zv3s9qhLdBN4E5zB4H4zy2CWz8",
    "nickname": "大罗",
    "sex": 0,
    "language": "zh_CN",
    "city": "深圳",
    "province": "广东",
    "country": "中国",
    "headimgurl": "http://wx.qlogo.cn/mmopen/47CicbLQOxtUCKpQukJicCFsJAhFVxaTCbILicibWC3fF2PLcsu8PSqv3LAOkh5fon5ElEGYzC0LA4fm3NceYnRGhQ/0",
    "subscribe_time": 1431066413,
    "remark": "",
    "groupid": 101
  },
  {
    "subscribe": 0,
    "openid": "o7Zv3s-qZgWxU8Kcze0lMilEa8a4",
    "nickname": "空空",
    "sex": 1,
    "language": "zh_CN",
    "city": "东城",
    "province": "北京",
    "country": "中国",
    "headimgurl": "http://wx.qlogo.cn/mmopen/ajNVdqHZLLDrmkDPv72sWDC9pC5icUGgNqKuls80nNQa9HibM96WqZkVkLFx18ACsicl3efN6GETLLAgxWd3aviaHw/0",
    "subscribe_time": 1432622962,
    "remark": "",
    "groupid": 0
  },
  {
    "subscribe": 1,
    "openid": "o7Zv3s5x1KWRYLoeCdopV_2VfeuY",
    "nickname": "丁丁",
    "sex": 0,
    "language": "zh_CN",
    "city": "浦东新区",
    "province": "上海",
    "country": "中国",
    "headimgurl": "http://wx.qlogo.cn/mmopen/PiajxSqBRaELUrmDwoaibukxoodjkehicfSzan0SfQLROSm03CGg29yL6XT2xeUjFqMI3nmcrfhPxL9ySiaKL4MN1A/0",
    "subscribe_time": 1432197923,
    "remark": "",
    "groupid": 0
  },
  {
    "subscribe": 1,
    "openid": "o7Zv3s2-dTvJhCqdTfV5E004u_SU",
    "nickname": "陈桥🐰",
    "sex": 1,
    "language": "zh_CN",
    "city": "",
    "province": "上海",
    "country": "中国",
    "headimgurl": "http://wx.qlogo.cn/mmopen/47CicbLQOxtUSRNAccBUR7JudkN7pNLbIhpOUzz5icmFzQRTtfqkdibMcicxShlV66T5jwtU3icF4ndYIDbLmMToc7g/0",
    "subscribe_time": 1444965474,
    "remark": "",
    "groupid": 0
  },
  {
    "subscribe": 1,
    "openid": "o7Zv3s6NTAiU3elVDXRyapJUoHMo",
    "nickname": "水长东 ™",
    "sex": 1,
    "language": "zh_CN",
    "city": "朝阳",
    "province": "北京",
    "country": "中国",
    "headimgurl": "http://wx.qlogo.cn/mmopen/s3NiblUuUDR7rwDibAao6WOrmc3PEdTnFu6F5HBqSPfQaABnrgV5lGAbhqfSzgria1nm8CXFvHicLQsYdMw9KVSLOg/0",
    "subscribe_time": 1435206773,
    "remark": "",
    "groupid": 0
  },
  {
    "subscribe": 1,
    "openid": "o7Zv3szwHllkIJnQQy1npd53PM8A",
    "nickname": "吴畏",
    "sex": 1,
    "language": "zh_CN",
    "city": "徐汇",
    "province": "上海",
    "country": "中国",
    "headimgurl": "http://wx.qlogo.cn/mmopen/PiajxSqBRaEIF2XFWtO7piaibcJ7RD4FkuVZicqWhhQLhn277hMHGABeesP53oFXAUhUm33o7LCg6eDedkJQsyawvw/0",
    "subscribe_time": 1447914819,
    "remark": "",
    "groupid": 101
  }
]